standard/ERCs/erc-6900.md

```diff
@@ -256,79 +279,202 @@ interface IAccountLoupe {
-    /// @param data Optional bytes array to be decoded and used by the module to setup initial module data for the modular account.
+    /// @dev Called by the modular account during `installExecution`.
+    /// @param data Optional bytes array to be decoded and used by the module to setup initial module data for the
+    /// modular account.
     function onInstall(bytes calldata data) external;
 
     /// @notice Clear module data for the modular account.
-    /// @dev Called by the modular account during `uninstallModule`.
-    /// @param data Optional bytes array to be decoded and used by the module to clear module data for the modular account.
+    /// @dev Called by the modular account during `uninstallExecution`.
+    /// @param data Optional bytes array to be decoded and used by the module to clear module data for the modular
+    /// account.
     function onUninstall(bytes calldata data) external;
 
-    /// @notice Run the pre user operation validation hook specified by the `entityId`.
-    /// @dev Pre user operation validation hooks MUST NOT return an authorizer value other than 0 or 1.
-    /// @param entityId An identifier that routes the call to different internal implementations, should there be more than one.
-    /// @param userOp The user operation.
-    /// @param userOpHash The user operation hash.
-    /// @return Packed validation data for validAfter (6 bytes), validUntil (6 bytes), and authorizer (20 bytes).
-    function preUserOpValidationHook(uint8 entityId, PackedUserOperation memory userOp, bytes32 userOpHash) external returns (uint256);
+    /// @notice Describe the metadata of the module.
+    /// @dev This metadata MUST stay constant over time.
+    /// @return A metadata struct describing the module.
+    function moduleMetadata() external pure returns (ModuleMetadata memory);
+}
+
+```
 
+#### `IValidationModule.sol`
+
+``````solidity
+interface IValidationModule is IModule {
     /// @notice Run the user operation validationFunction specified by the `entityId`.
-    /// @param entityId An identifier that routes the call to different internal implementations, should there be
-    /// more than one.
+    /// @param entityId An identifier that routes the call to different internal implementations, should there
+    /// be more than one.
     /// @param userOp The user operation.
     /// @param userOpHash The user operation hash.
     /// @return Packed validation data for validAfter (6 bytes), validUntil (6 bytes), and authorizer (20 bytes).
-    function userOpValidationFunction(uint8 entityId, PackedUserOperation calldata userOp, bytes32 userOpHash)
+    function validateUserOp(uint32 entityId, PackedUserOperation calldata userOp, bytes32 userOpHash)
         external
         returns (uint256);
 
-    /// @notice Run the pre runtime validation hook specified by the `entityId`.
+    /// @notice Run the runtime validationFunction specified by the `entityId`.
```

Comment:
Should we prefer to spell it out as "validation function" instead of the condensed "validationFunction"?

Reply:
we should. Updated all the refs in comments and docs.